kratos/includes/properties.h

```diff
@@ -277,31 +290,121 @@ class Properties : public IndexedObject
         return result_key;
     }
 
+    /**
+     * @brief This method returns the number of subproperties
+     * @return The current number of subproperties
+     */
+    std::size_t NumberOfSubproperties()
+    {
+        return mSubPropertiesList.size();
+    }
+
+    /**
+     * @brief This method insert a new property into the list of subproperties
+     * @param pNewSubProperty The new property to be added
+     */
+    void AddSubProperty(Properties::Pointer pNewSubProperty)
+    {
+        mSubPropertiesList.insert(mSubPropertiesList.begin(), pNewSubProperty);
+    }
+
+    /**
+     * @brief This method gets the subproperty from the index corresponding to the property id
+     * @param SubPropertyIndex The index of the subproperty to be get
+     * @return The pointer to the subproperty of interest
+     */
+    Properties::Pointer pGetSubProperty(IndexType SubPropertyIndex)
+    {
+        return mSubPropertiesList(SubPropertyIndex);
```

Comment:
Before this return shall we add a check if the `mSubPropertiesList` is not empty ?
Also at line 328 !?

Reply:
OK